src/context/notes-context.js

```diff
@@ -0,0 +1,43 @@
+import { createContext, useContext, useReducer } from "react";
+
+const NotesContext = createContext();
+const useNotesContext = () => useContext(NotesContext);
+
+const notesReducer = (state, { type, payload }) => {
+  switch (type) {
+    case "USER":
+      return { ...state, user: payload };
+    case "NOTES":
+      return { ...state, user: { ...state.user, notes: payload } };
+    case "TITLE":
+      return { ...state, title: payload };
+    case "DESCRIPTION":
+      return { ...state, description: payload };
+    case "CLEAR_NOTES_INPUTS":
```

Comment:
suggestion: would suggest not to dispatch the action for each field, as this will update Context and state, so there will be a re render of the child components from reactjs.org 
![image](https://user-images.githubusercontent.com/98606563/163552504-146b6233-16da-4c10-8bcd-1c1bb6a18aa9.png)

Reply:
will read about this.